src/gpu/amd/miopen_binary.cpp

```diff
@@ -40,7 +40,7 @@ status_t miopen_binary_t::execute(const exec_ctx_t &ctx) const {
         auto arg_scale1
                 = CTX_IN_SYCL_MEMORY(DNNL_ARG_ATTR_SCALES | DNNL_ARG_SRC_1);
 
-        compat::host_task(cgh, [=](const compat::interop_handle &ih) {
+        compat::host_task(cgh, [=, this](const compat::interop_handle &ih) {
```

Comment:
Shouldn't this use `WA_THIS_COPY_CAPTURE`, just like https://github.com/oneapi-src/oneDNN/pull/2506 to avoid the original issue mentioned in MFDNN-13026?

Reply:
Copy-pasted from existing Nvidia backend code base.
Code enhancements would be appreciated for sure.

Reply:
If this WA is required for build to succeed this is not a code enhancement :)

Reply:
It's working for the other backend, why wouldn't it for this?
If it wouldn't for some reason I'll submit another PR.

Reply:
Thanks for fixing this. So it seems for the NVidia backend we can use `[=, this]` because the warnings were disabled in [CMake](https://github.com/oneapi-src/oneDNN/blob/main/cmake/SYCL.cmake#L81) with `-Wno-c++20-extensions`. On AMD the warnings are emitted and it fails due to `-Werror`.
I'm pretty sure we can't use capture by reference to capture buffer's accessors in SYCL. It could work for USM but the paths for USM and buffers are the same.
I think we could improve the situation by removing `-Wno-c++20-extensions` and using `WA_THIS_COPY_CAPTURE` when needed.